packages/routers/src/TabRouter.tsx

```diff
@@ -95,6 +93,13 @@ const changeIndex = (
     history = state.history
       .filter((it) => (it.type === 'route' ? it.key !== currentKey : false))
       .concat({ type: TYPE_ROUTE, key: currentKey });
+  } else if (backBehavior === 'initialRoute') {
+    const currentKey = state.routes[index].key;
+
+    history = [state.history[0]];
```

Comment:
History should be populated inside the `getHistory` function for it to work correctly. There are changes to history in this function when `backBehavior` is `history`, because in that case, `history` is coupled to the `index` change.